### DISCUSSION TITLE
Unable to get both floppy drive & Gotek to work together

### DISCUSSION THREAD

**Discussion by UFO-OS9** [keirf/flashfloppy#743] [Q&A] [answered]
Hello... I'm looking for a bit of help if possible ?
I use an old  Windows 95 PC with two floppy drives to write & duplicate floppy's. The problem is that the disks I write are not for  DOS nor Windows they are for an old OS9 system whereby the manufacturer provided a utility to bit write the floppy's from an image file via a non NT windows PC which takes forever !

So after discovering the Gotek, I thought that I could use the original  image, copy it to the Gotek USB .... and away we go.  NO !
If I try replacing the A: drive with the Gotek I get my first problem (The software utility is set by default to write only to Drive A: which is not adjustable) The PC can't identify which drive is which if I click on B: drive both the Gotek & the B: drives light up for reading but neither read. If I just connect Drive A: it connects but the the utility stalls after formatting the selected floppy on the USB stick (basically, the utility inspects the floppy then formats it, then writes it).  

I then thought, despite the setbacks, if I use a bit copier utility (Anadisk) on the PC, copy from floppy drive B: to Gotek on Drive A: it should work ?? ..NO ! because I'm back to my drive read problem.

I have reversed Floppy ribbon cable drive allocator connections (cable has twist in )
Can you actually use a Gotek along side another floppy ie Gotek on A: Floppy drive on B:  ?

What I do know:
Latest Gotek pcb with  firmware v3.36
I'm  currently using just S0 jumper as it works perfectly with Windows/DOS as a single floppy emulator
FF.cfg is set to:  interface = ibmpc   have tried - host = pc-dos
I've tried winimage to "inject" my Os9 image but that doesn't like the image, as its 1.6mb which is to big for 1.44 floppy

Any help would be most appreciated.... Thanks for reading !

**Comment by keirf** [comment]
First things first, the correct jumper setting is S1. And the correct interface setting will be ibmpc. With S0 jumpered unexpected behaviours can be expected.

**Reply by UFO-OS9** [reply]
Thank you keirf for answering, the jumper set at S1,  resolves the drive read issues ..... most appreciated ! Thanks !
But I still have my OS9 write problem hanging the Gotek.
If I swap over the Gotek to drive B: & assign my old floppy drive to drive A: (just to test that it still works as it used to), will I have to reset the jumper on the Gotek to another position or should it still be OK at S1 ?  is there any  documentation regarding the jumpers as I cannot find this stuff anywhere? ... I would rather not, wish to waste your time asking silly questions.
It is appreciated tho !

**Reply by keirf** [reply]
PC drives are always S1. Drive letter is decided by cable (there is a twist in the cable which changes drive letter downstream of the twist).

Is your Gotek with 415 chip? These have little RAM for buffering writes. It is explained in the flashfloppy wiki in the gotek compatibility section.

**Reply by UFO-OS9** [reply]
Thanks again .. Yes I have 415 chip .. so I'll take a look at the text & do a bit more experimenting .

**Reply by keirf** [reply]
If you want to try a 435 Gotek I sell them in UK. For other countries I sometimes ship internationally or I may know a local seller.

**Reply by UFO-OS9** [reply]
After reading....That does sound like ....exactly whats going on here

**Reply by UFO-OS9** [reply]
Its ironic...I just ordered another unit thinking I had a faulty device, but its going to be a 415...  I will try the file HXCSDFE.CFG business but drop me a line with your details

**Reply by keirf** [reply]
You can email me at keir.xen@gmail.com as listed on my GitHub profile.

**Reply by UFO-OS9** [reply]
Just installed  the 435 Gotek.... but I have the same issue when trying to write OS9 floppy's via a Windows 95 PC using a manufacturers utility (The software utility is set by default to write only to Drive A: which is not adjustable). just to prove that all is OK, using my "old" method I swapped the Gotek for the old floppy drive & tried the same procedure and it works perfectly, which means my set up is good.
So its something to do with the Gotek/ flash floppy configuration / capability.

Basically, the utility inspects the floppy then formats it, then writes it.  It inspects OK, it formats OK but as soon as it starts to write I get an "input" error  ??
If I use the Gotek to read & write files via Windows its works as expected.
Any help would be most appreciated.... Thanks for reading !

**Reply by keirf** [reply]
What USB stick are you using? Have you tried another? Does format happen so fast it might be formatting back to back and not verifyinh each track as it formats it?
Are you using HFE or IMG files? HFE I assume. Do you know if the disks being written are double density (up to 1MB capacity) or high density (up to 2MB capacity)?

**Reply by UFO-OS9** [reply]
Ive been using a 8GB generic usb stick, but have just tried  a 14GB Sandisk results are the same, formatting is slow almost the same as a floppy, I have been using .IMG files, just tried an unformatted .HFE file but that hangs the Gotek when attempting to format it.
The floppys I use are 2HD so I assume that they are High density. One thing that might help, is that the image file used to write the disks  is 1.6Kb.  getting back to basics, can you confirm if the jumper at J5 - JA is required or not ? (I have tried both with & without)

**Reply by keirf** [reply]
That jumper is a spare. It is actually across two ground pins.

**Reply by keirf** [reply]
What HFE file did you use? IMG may not work. They are "fixed format" and if OS9 uses a different unexpected track format, the format operation will simply not work.

**Reply by keirf** [reply]
It probably doesn't matter how the HFE is formatted, as it sounds like your disk creation process completely rewrites the disk anyway. Which of the host types within that zip file did you pick? Did you know whether OS9 uses double- or high-density disks?

Is it actually a 880k GCR you're trying to create? That's hard due to variable rpm across zones of the disk. I don't actually think that is supported at all! But then you couldn't probably write those from a PC in any case.

If it's a PC-style MFM high-density disk, then that should for sure be possible.

**Reply by UFO-OS9** [reply]
I used this: https://github.com/keirf/FF_Images/releases/download/latest/FF_Images.zip
I tried all four images from the "unformatted" folder......  
I physically use : BASF  2 HD High Density - Double Sided - Formatted 1.44 MB. floppy's, only because I have/had  box's of them from the manufacturer years ago, but they work fine

**Reply by keirf** [reply]
What's the storage capacity of the disks in OS9?

**Reply by UFO-OS9** [reply]
1.44 Mb.....   I've just looked at the bat file, that starts the OS9 writing, but there is nothing much in there,  other than write path - diskrw A: W -t=80 -s=10 -b=1024 -f=E5 -n   Does that info, throw any light on the matter ?

**Reply by keirf** [reply]
I think that's writing 10 1kB sectors per track for a total capacity of 1.6MB.

So you would want a high density HFE for sure (approx 4MB in size).

Or we need to work out the exact disk geometry so we can set you up a configuration using IMG files.

You say that using HFE "hangs"? What does that look like?

**Reply by UFO-OS9** [reply]
That makes sense because the size of the image file it uses to write from is 1.6Mb.
regarding the "hanging" when I tried to quick format the HFE image(s) from the unformatted folder, windows threw  first message cannot do quick format press OK to Full format... then it stalls & just greys out the format window. & hangs for about 10 minutes.
But all that is prior to any OS9 writing, I should be able to format those images, despite the OS9 stuff re-formatting them.

I'll look out for a  high density HFE  image (approx 4MB in size) & give it a try...  Thanks for your time !

**Reply by keirf** [reply]
`high_density.hfe` contained within the FF_Image.zip should be the right one. Windows will simply be formatting it to 1.44MB.

Here is one alreadfy formatted to 1.44MB. Windows might accept this without trying to "full format" it.

[hd.zip](https://github.com/keirf/flashfloppy/files/10406350/hd.zip)

Unfortunately full track formatting/copying at high speed is hard with high-density HFE as we only have a full-speed USB connection to the USB stick and this seriously limits the write speed. High-density IMG is much easier to deal with but we need to know the disk geometry and configure that for FlashFloppy via an IMG.CFG file.

**Reply by UFO-OS9** [reply]
Thanks.. I don't know how you have the patience.....   just tried hd.HFE  first going straight to the OS9 write utility.... but just the same formatted it went to write it & input error 
Just out of curiosity I tried a windows quick format which it did, went on to the OS9 write..... exactly the same..

**Reply by keirf** [reply]
What if you use Anadisk?

**Reply by keirf** [reply]
Finally, what would be really useful is a raw dump of an actual 1.6MB OS9 disk. Then you can use your 1.6MB data file directly as an IMG file. You wouldn't even need the copying step!

**Reply by keirf** [reply]
To this end you could try renaming your 1.6mb file with an IMG suffix and use it directly as a disk image. You would need to try `host=akai` in FF.CFG

**Reply by UFO-OS9** [reply]
Re-naming the original image was actually the very first task I ever did with the Gotek. I have just retried it, using host+akai but still get same  can't read disk error
Re Anadisk, it stalls with "cannot read disk"
Finally ... ...  .
I tried a 1.6 Mb blank .IMG  file ...  .. when the OS9 writer attempted to format it, it returned "Drive not ready" error.

Thanks !

**Reply by keirf** [reply]
Is the OS9 system a different machine than the PC? Does it also use PC compatible disk drives (which would mean that jumper S1 would be correct on that system too)?

The akai geometry is not guaranteed correct. We could try some variants. Another option is send me a OS9 disk and I can view its - geometry. Or buy a Greaseweazle and take a raw dump of an OS9 disk and send that to me.

**Reply by UFO-OS9** [reply]
The OS9 is a different Machine than the PC & they both use standard Floppy drives  but here's a twist with the Gotek jumper set to J1
then installed to the OS9 machine the Gotek does not acknowledge a read request during Start up, but if I change the jumper to S0 The read led lights it still does not load the image... but in earlier post you advised not to use jumper S0 ?

**Reply by keirf** [reply]
You should see the green LED light when the drive is accessed. Does the OS9 system work with a regular unmodified PC drive (which would also work with your PC system)? If so then S1 really should be the correct jumper position. All PC drives respond on S1 select line (floppy interface pin 12).

**Reply by UFO-OS9** [reply]
I need to do a bit more testing with the different OS9 systems  that I have, but I'm sure its working now  !!! ...   Jumper S0 is definitely required for use on my present OS9 unit, as its the only setting that strikes the access led during boot.

I've tried that many different configurations I'm not 100% sure, what I have actually changed in the FF.CFG, as I have only been using single line entries to the FF.CFG while recently testing.  Is there a way to extract the flash floppy configuration settings from the gotek, in order to review/confirm the current settings, otherwise I will need to start over.

**Reply by keirf** [reply]
You can't extract the configuration unfortunately. You will need to factory reset and then build up your FF.CFG line by line.

**Reply by keirf** [reply]
Probably only interface= and host= lines will matter.

**Comment by UFO-OS9** [comment, marked as answer]
The Solution.....
After all the dickin about...  .. Finally the Gotek runs perfectly on three totally different OS9 systems, using the original manufacturers images. no copying, formatting & writing utilities required:

  **Jumper set to: J0**
```
interface = ibmpc
host = akai
```
Thanks Keir for all your help !

**Reply by keirf** [reply]
I'll add it to the wiki. This is Microware OS-9 we're talking about?

**Reply by UFO-OS9** [reply]
Correct..

**Reply by keirf** [reply]
https://github.com/keirf/flashfloppy/wiki/Host-Platforms#microware
